backend/src/apiserver/resource/resource_manager.go

```diff
@@ -195,616 +203,989 @@ func (r *ResourceManager) ArchiveExperiment(ctx context.Context, experimentId st
 			opts, err = list.NewOptionsFromToken(newToken, 50)
 			if err != nil {
 				return util.NewInternalServerError(err,
-					"Failed to create list jobs options from page token when archiving experiment. ")
+					"Failed to create list jobs options from page token when archiving experiment %v", experimentId)
 			}
 		}
 	}
 	return r.experimentStore.ArchiveExperiment(experimentId)
 }
 
+// Un-archives the experiment with the given id.
 func (r *ResourceManager) UnarchiveExperiment(experimentId string) error {
 	return r.experimentStore.UnarchiveExperiment(experimentId)
 }
 
-func (r *ResourceManager) ListPipelines(filterContext *common.FilterContext, opts *list.Options) (
-	pipelines []*model.Pipeline, total_size int, nextPageToken string, err error) {
-	return r.pipelineStore.ListPipelines(filterContext, opts)
+// Returns a list of pipelines.
+func (r *ResourceManager) ListPipelines(filterContext *model.FilterContext, opts *list.Options) ([]*model.Pipeline, int, string, error) {
+	pipelines, total_size, nextPageToken, err := r.pipelineStore.ListPipelines(filterContext, opts)
+	if err != nil {
+		err = util.Wrapf(err, "Failed to list pipelines with context %v, options %v", filterContext, opts)
+	}
+	return pipelines, total_size, nextPageToken, err
+}
+
+// TODO(gkcalat): consider removing after KFP v2 GA if users are not affected.
```

Comment:
You are  right. The TODOs are for consideration. Removal of the functions will not necessarily lead to a change of behavior. For example, `ListPipelineV1` joins `pipelines` and `pipeline_versions` tables. While this is efficient, we may (or may not) be okay to split this call into two separate `ListPipelines` and `ListPipelineVersions`. This way, we could achieve `resource` to be API version agnostic, while handling API version-specific logic at the `server` layer. It will clearly cause a performance degradation, which is too big right now, but as KFP v2 adoption progresses, we may return to this item in the future. I agree, this could happen beyond 1 year after the GA.